internal/restic/cache.go

```diff
@@ -35,3 +38,10 @@ type Cache interface {
 	// Has returns true if the file is cached.
 	Has(Handle) bool
 }
+
+// CachedBackend enhances the Backend interfaces to work with caches
```

Comment:
There's no need for another [interface with only one implementation](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a). I know this is how restic is currently structured in some places, but I'd argue for "return structs, accept interfaces" in new code. Also, the methods aren't documented.